src/components/common/TmSessionExplore.vue

```diff
@@ -170,18 +178,24 @@ export default {
         return false
       }
     },
-    isANetworkAddress(param) {
-      const selectedNetwork = this.addressPrefixes.find(
-        ({ id }) => id === this.network
+    async selectNetworkByAddress(address) {
+      let selectedNetwork = this.addressPrefixes.filter(({ address_prefix }) =>
+        address.startsWith(address_prefix)
       )
       // handling query not loaded yet or failed
-      if (!selectedNetwork) return false
-
-      if (param.startsWith(selectedNetwork.address_prefix)) {
-        return true
+      if (!selectedNetwork) {
+        console.error("Connecting to the selected network failed")
+        return
+      }
+      // handling when there are both mainnet and testnet networks
+      if (selectedNetwork.length > 1) {
+        selectedNetwork = selectedNetwork.filter(({ id }) =>
+          this.testnet ? id.slice(-7) === `testnet` : id.slice(-7) === `mainnet`
+        )[0]
       } else {
-        return false
+        selectedNetwork = selectedNetwork[0]
```

Comment:
Do we need the else?

Reply:
Yes. For all the networks that only have testnet (the vast majority right now).

I think I should change the selectedNetwork to plural to make it more clear.

Reply:
but if the length of `selectedNetwork` is 1 you can still use the same logic of filtering it. There should be a handling if there is no testnet or mainnet for the network but the user selected it.